### DISCUSSION TITLE
[Copy plain Text] Regex replace help

### DISCUSSION THREAD

**Discussion by rosgr100** [erosman/support#549] [Q&A] [answered]
Hi, i'm trying to create a regex which will bring the second line to the top while at the same time, it will retain a space between the text.

Example

```
Text
Second line text
```

To

`Text Second line text`

How would i do that?

Thanks

**Comment by erosman** [comment, marked as answer]
## Basic

Here is a JavaScript `replace()` example:

```js
const str = `Text
Second line text`;
const newStr = str.replace(/\r?\n/, ' ');
// Text Second line text
```

If you want more than one line to be replaced, then:

```js
const newStr = str.replace(/\r?\n/g, ' ');
```

In **Copy PlainText**:

**Find:** `/\r?\n/`
**Replace:** *put a single space*

## Advanced
There are more possible variations which might require adjustment to the regex e.g.

```
Text


Second line text
```

Or ...
```
Text (space)(space)(space)
Second line text
```

This is a more comprehensive regx:

```js
const str = `Text
Second line text`;
const newStr = str.replace(/\s*[\r\n]+\s*/g, ' ');
```

**Comment by rosgr100** [comment]
Thank you @erosman for all the help and the effort you've to explain so thoroughly every sinlge detail or case.

I tried some different patterns but none of them worked, yours in the other hand is working perfectly.

Thanks again!
